opentracing-spring-web-handler-interceptor/src/test/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptorTest.java

```diff
@@ -0,0 +1,162 @@
+        mockServer.expect(MockRestRequestMatchers.requestTo(url))
+                .andExpect(MockRestRequestMatchers.method(HttpMethod.GET))
+                .andRespond(new ResponseCreator() {
+                    @Override
+                    public ClientHttpResponse createResponse(ClientHttpRequest request) throws IOException {
+                        MockClientHttpResponse response =
+                                new MockClientHttpResponse(new byte[1], HttpStatus.OK);
+
+                        response.getHeaders().add("traceId", request.getHeaders()
+                                .getFirst("traceId"));
+                        response.getHeaders().add("spanId", request.getHeaders()
+                                .getFirst("spanId"));
+                        return response;
+                    }
+                });
+
+        ResponseEntity<String> responseEntity = restTemplate.getForEntity(url, String.class);
+
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        Assert.assertEquals(1, mockSpans.size());
+        Assert.assertEquals(mockSpans.get(0).context().traceId(),
+                Long.parseLong(responseEntity.getHeaders().getFirst("traceId")));
+        Assert.assertEquals(mockSpans.get(0).context().spanId(),
+                Long.parseLong(responseEntity.getHeaders().getFirst("spanId")));
+    }
+
+    @Test
+    public void testErrorUnknownHostException() {
+        String url = "http://nonexisting.example.com";
+        try {
+            RestTemplate restTemplate = new RestTemplate();
+            restTemplate.setInterceptors(Collections.<ClientHttpRequestInterceptor>singletonList(
+                    new TracingRestTemplateInterceptor(mockTracer)));
+            restTemplate.getForEntity(url, String.class);
+        } catch (ResourceAccessException ex) {
+            //ok UnknownHostException
+        }
+
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        Assert.assertEquals(1, mockSpans.size());
+
+        MockSpan mockSpan = mockSpans.get(0);
+        Assert.assertEquals(5, mockSpan.tags().size());
+        Assert.assertEquals(RestTemplateSpanDecorator.StandardTags.COMPONENT_NAME,
+                mockSpan.tags().get(Tags.COMPONENT.getKey()));
+        Assert.assertEquals(Tags.SPAN_KIND_CLIENT, mockSpan.tags().get(Tags.SPAN_KIND.getKey()));
+        Assert.assertEquals("GET", mockSpan.tags().get(Tags.HTTP_METHOD.getKey()));
+        Assert.assertEquals(url, mockSpan.tags().get(Tags.HTTP_URL.getKey()));
+        Assert.assertEquals(Boolean.TRUE, mockSpan.tags().get(Tags.ERROR.getKey()));
+
+        Assert.assertEquals(1, mockSpan.logEntries().size());
+        Assert.assertEquals(2, mockSpan.logEntries().get(0).fields().size());
+        Assert.assertEquals(Tags.ERROR.getKey(), mockSpan.logEntries().get(0).fields().get("event"));
+        Assert.assertNotNull(mockSpan.logEntries().get(0).fields().get("error.object"));
+    }
+}
```

Comment:
:+1: good point, I think it should be responsibility of an injector/propagator configured on `Tracer`
* https://github.com/opentracing/opentracing-java/blob/master/opentracing-impl/src/main/java/io/opentracing/impl/AbstractTracer.java#L33
* https://github.com/opentracing/opentracing-java/blob/master/opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java#L49